src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -192,4 +224,15 @@ async def construct_workunits_callback_factories(
 
 
 def rules():
-    return [QueryRule(WorkunitsCallbackFactories, (UnionMembership,)), *collect_rules()]
+    return [
+        QueryRule(WorkunitsCallbackFactories, (UnionMembership,)),
+        QueryRule(Targets, (Addresses,)),
+        QueryRule(
+            Addresses,
+            (
+                Specs,
+                OptionsBootstrapper,
+            ),
```

Comment:
Formatting nit: Black only formatted this on multiple lines because you had a stray comma at the end before running `./pants fmt`, so it interpreted that you wanted everything split over a new line. You can remove the trailing commas on both 234 and 235, and Black should fix this to be single line.

(Not required, more an FYI.)